src/components/MenuProductModal/MenuProductModal.tsx

```diff
@@ -0,0 +1,125 @@
+import { ReactElement, useCallback, useState } from 'react';
+import { ProductDto } from 'dtos/ProductDto';
+import { Button } from 'components/Button/Button';
+import { useLocalOrder } from 'context/OrderContext';
+import PlaceHolderImage from 'assets/burger_figma.png';
+
+interface Props {
+  product: ProductDto;
+  showModal: boolean;
+  disableShow: (isShow: boolean) => void;
+}
+
+export const MenuProductModal = ({ product, showModal, disableShow }: Props): ReactElement => {
+  const { addToLocalOrder, removeProduct, localOrder } = useLocalOrder();
+
+  const checkQuantity = (product: ProductDto): number => {
+    const existingItem = localOrder.lineItems.find((item) => item.product.id === product.id);
+    return existingItem ? existingItem.quantity : 1;
+  };
+
+  const [counter, setCounter] = useState(checkQuantity(product));
+  const increase = (): void => {
+    setCounter((counter) => counter + 1);
+  };
+
+  const decrease = (): void => {
+    if (counter) {
+      setCounter((counter) => counter - 1);
+    }
+  };
+
+  const addToOrder = useCallback(() => {
+    addToLocalOrder(product, counter);
+  }, [addToLocalOrder, product, counter]);
+
+  const allergens = ['Milk', 'Peanuts', 'Celery', 'Soy', 'Gluten'];
+
+  return (
+    <>
+      {showModal ? (
+        <>
+          <div
+            className="fixed inset-0 z-50 flex items-center justify-center overflow-x-auto overflow-y-auto
```

Comment:
I don't think we should be setting the z-index nor the inset. It should be something like 'fixed top-0 left-0 flex h-full w-full'